### PR TITLE
[IES-410] Export linting config

### PR DESCRIPTION
**Pull request by theRossFromFriends**
https://capmo-team.atlassian.net/browse/IES-410

PR aims to export linting config from this repository.

### CODE REVIEW

File: Linting and Styling/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { prettierrc } = require('./prettierrc')
+const { esLintrc } = require('./eslintrc')
+
+const prettier = {
+  prettierrc
```
Comment (theRossFromFriends) [inline Linting and Styling/index.js:5]: Possibly we would put inside the `prettier` object also the prettierignore (the ticket description says 'place the ignore and RC under one file and export them both for easy access'), I am not sure however if it is possible.

**Review by vahid-capmo — CHANGES_REQUESTED** [review]
Can we consider separating eslint stuff from prettier stuff into their own folder? and folder name should not contain spaces

File: Linting and Styling/eslintrc.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{
+const esLintrc = {
```
Comment (vahid-capmo) [inline Linting and Styling/eslintrc.js:1]: Can we rename it to better reflect it's purpose? like `eslintConfig`?

File: Linting and Styling/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+}
+
+module.exports = {
+  prettier, esLintrc
```
Comment (vahid-capmo) [inline Linting and Styling/index.js:9]: Can we make this repo use prettier to format the code inside it's own repo?

File: Linting and Styling/prettierrc.js [hunk 4]
```diff
@@ -1,7 +1,11 @@
-{
+const prettierrc = {
```
Comment (vahid-capmo) [inline Linting and Styling/prettierrc.js:1]: Can we rename it to `prettierConfig`?

File: index.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { esLintrc, prettier, }  from './linting_styling'
+
+module.exports = {
+ esLintrc,
+ prettier
+}
```
Comment (vahid-capmo) [inline index.js:6]: Could use new line

File: Linting and Styling/index.js (hunk 3 above)
Comment (GregoryEDev) [reply Linting and Styling/index.js:9]: I have done this in my PR

File: Linting and Styling/prettierrc.js (hunk 4 above)
Comment (GregoryEDev) [inline Linting and Styling/prettierrc.js:1]:
```suggestion
const prettierConfig = {
```

File: Linting and Styling/prettierrc.js [hunk 6]
```diff
@@ -1,7 +1,11 @@
...
   "semi": false,
   "trailingComma": "none",
   "singleQuote": true,
   "printWidth": 80,
   "jsxSingleQuote": true
 }
+
+module.exports = {
+  prettierrc
```
Comment (GregoryEDev) [inline Linting and Styling/prettierrc.js:10]:
```suggestion
  prettierConfig
```

File: README.md [hunk 7]
```diff
@@ -4,6 +4,32 @@ Basic, extendable configuration for Product and Engineering team
...
+// "extends": [ "airbnb-typescript" ]
+};
+```
+
+Prettier
+
+Create a `.prettierrc.js` file in the root of the project with the following code
+
+```
+const { prettierrc } = require('@capmo/config/prettier');
```
Comment (GregoryEDev) [inline README.md:28]:
```suggestion
const { prettierConfig } = require('@capmo/config/prettier');
```

File: README.md [hunk 8]
```diff
@@ -4,6 +4,32 @@ Basic, extendable configuration for Product and Engineering team
...
+
+module.exports = { ...prettierrc };
```
Comment (GregoryEDev) [inline README.md:30]:
```suggestion
module.exports = { ...prettierConfig };
```

File: index.js (hunk 5 above)
Comment (GregoryEDev) [inline index.js:6]:
```suggestion
}

```

File: Linting and Styling/index.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+const { prettierrc } = require('./prettierrc')
```
Comment (GregoryEDev) [inline Linting and Styling/index.js:1]:
```suggestion
const { prettierConfig } = require('./prettierrc')
```

File: Linting and Styling/index.js (hunk 1 above)
Comment (GregoryEDev) [inline Linting and Styling/index.js:5]:
```suggestion
  prettierConfig
```

File: Linting and Styling/eslintrc.js (hunk 2 above)
Comment (GregoryEDev) [inline Linting and Styling/eslintrc.js:1]:
```suggestion
const eslintConfig = {
```

File: Linting and Styling/eslintrc.js [hunk 10]
```diff
@@ -48,3 +48,7 @@
     ]
   }
 }
+
+module.exports = {
+  esLintrc
```
Comment (GregoryEDev) [inline Linting and Styling/eslintrc.js:53]:
```suggestion
  eslintConfig
```

File: Linting and Styling/index.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+const { esLintrc } = require('./eslintrc')
```
Comment (GregoryEDev) [inline Linting and Styling/index.js:2]:
```suggestion
const { eslintConfig } = require('./eslintrc')
```

File: Linting and Styling/index.js (hunk 3 above)
Comment (GregoryEDev) [inline Linting and Styling/index.js:9]:
```suggestion
  prettier, eslintConfig
```

File: README.md [hunk 12]
```diff
@@ -4,6 +4,32 @@ Basic, extendable configuration for Product and Engineering team
 
 # Linting and Styling
 
+ESLint
+
+Create a new `.eslintrc.js` file in the root of the project
+
+```
+const { esLintrc } = require('@capmo/config');
```
Comment (GregoryEDev) [inline README.md:12]:
```suggestion
const { eslintConfig } = require('@capmo/config');
```

File: README.md [hunk 13]
```diff
@@ -4,6 +4,32 @@ Basic, extendable configuration for Product and Engineering team
...
+
+module.exports = {
+ ...esLintrc
```
Comment (GregoryEDev) [inline README.md:15]:
```suggestion
 ...eslintConfig
```

File: index.js [hunk 14]
```diff
@@ -0,0 +1,6 @@
+import { esLintrc, prettier, }  from './linting_styling'
+
+module.exports = {
+ esLintrc,
```
Comment (GregoryEDev) [inline index.js:4]:
```suggestion
 eslintConfig,
```

File: index.js [hunk 15]
```diff
@@ -0,0 +1,6 @@
+import { esLintrc, prettier, }  from './linting_styling'
```
Comment (GregoryEDev) [inline index.js:1]:
```suggestion
import { eslintConfig, prettier, }  from './linting_styling'
```

**Review by GregoryEDev — APPROVED** [review]

File: index.js [hunk 16]
```diff
@@ -0,0 +1,6 @@
+import { eslintConfig, prettier, }  from './linting_styling'
```
Comment (vahid-capmo) [inline index.js:1]: Unnecessary comma after prettier import

File: README.md [hunk 17]
```diff
@@ -4,6 +4,32 @@ Basic, extendable configuration for Product and Engineering team
...
+// "extends": [ "airbnb-typescript" ]
+};
+```
+
+Prettier
+
+Create a `.prettierrc.js` file in the root of the project with the following code
+
+```
+const { prettierConfig } = require('@capmo/config/prettier');
```
Comment (vahid-capmo) [inline README.md:28]: Does this line actually work?

**Review by vahid-capmo — COMMENTED** [review]
What is your point of view on folder name `Linting and Styling`?
@GregoryEDev

File: index.js (hunk 16 above)
Comment (GregoryEDev) [inline index.js:1]:
```suggestion
import { eslintConfig, prettier }  from './linting_styling'
```

File: Linting and Styling/index.js [hunk 18]
```diff
@@ -0,0 +1,10 @@
+const { prettierConfig } = require('./prettierrc')
+const { eslintConfig } = require('./eslintrc')
+
+const prettier = {
+  prettierConfig
+}
+
+module.exports = {
+  prettier, eslintConfig
+}
```
Comment (GregoryEDev) [inline Linting and Styling/index.js:10]:
```suggestion
const { prettierConfig } = require('./prettierrc')
const { eslintConfig } = require('./eslintrc')

module.exports = {
  prettierConfig, eslintConfig
}
```

File: README.md (hunk 17 above)
Comment (GregoryEDev) [inline README.md:28]:
```suggestion
const { prettierConfig } = require('@capmo/config');
```

**Review by vahid-capmo — APPROVED** [review]

**Pull request merged**